python/src/worker/tasks/marker_heatmap.py

```diff
@@ -42,52 +42,52 @@ def _format_request(self):
     def compute(self):
         request = self._format_request()
 
-        r = requests.post(
+        response = requests.post(
             f"{config.R_WORKER_URL}/v0/runMarkerHeatmap",
             headers={"content-type": "application/json"},
             data=json.dumps(request),
         )
-        # raise an exception if an HTTPError if one occurred because otherwise r.json() will fail
-        r.raise_for_status()
-        resultR = r.json()
-        truncatedR = resultR["truncatedExpression"]
-        resultR = resultR["rawExpression"]
-        result = {}
+        # raise an exception if an HTTPError if one occurred because otherwise response.json() will fail
+        response.raise_for_status()
+        result = response.json()
+        self.set_error(result)
+        if self.error:
+            return self._format_result(None)
+
+        truncatedExpression = result["truncatedExpression"]
+        rawExpression = result["rawExpression"]
+        work_result = {}
```

Comment:
Why is it called work_result here and result everywhere else? I'd go with the same option everywhere.

Reply:
changed to `result`. The previous `result` variable is changed to `response_json`